(MODULES-7814) Ignore nil values when parsing commands. Fix for: Could not evaluate: undefined method strip! for nil:NilClass

Lavinia-Dan opened the pull request:
While investigating an issue for a nightly build we came across this error:
`Error: /Stage[main]/Main/Tomcat::Config::Server[tomcat8-jsvc]/Augeas[server-/opt/apache-tomcat/tomcat8-jsvc]: Could not evaluate: undefined method "strip!" for nil:NilClass`
After making this modification locally first and running the tests again, everything works fine. 
There was already a logic in place to check whether the data was nil or not, however this should have been evaluated before the strip operation.

joshcooper commented on lib/puppet/provider/augeas/augeas.rb:
```diff
@@ -79,8 +79,8 @@ def parse_commands(data)
     data.each do |line|
-      line.strip!
       next if line.nil? || line.empty?
+      line.strip!
```
Thanks for your contribution @Lavinia-Dan. The change makes sense but isn't quite right, because we need to `strip!` before checking if the line is `empty?`. So maybe something like:

```ruby
next if line.nil?
line.strip!
next if line.empty?
```

Lavinia-Dan replied:
Thank you. Fixed now

joshcooper commented:
Could you update the commit summary with the ticket number, eg. "(MODULES-7814) Ignore nil values when parsing commands"

Also could you add a unit test for this, somewhere in the ["command parsing" example group](https://github.com/puppetlabs/puppetlabs-augeas_core/blob/master/spec/unit/provider/augeas/augeas_spec.rb#L48)?

Lavinia-Dan commented:
I've updated the message to match the ticket. And I've added a unit test. I was unsure whether to test it only for nil and have something like 
`commands = [nil]  
expect(tokens.size).to eq(1)` or if I should multiple options for the command. 
I figured since this is for command parsing I should use multiple options and check that only the nil is ignored, and the other options are properly processed for data that contains both nil and strings. 
I hope this is fine now.

joshcooper commented:
Looks good. Could you squash the commits?

davidmalloncares commented:
Hey @joshcooper, looks like the commits are squashed already? Is it good to merge?
